code/modules/client/preference_setup/preference_setup.dm

```diff
@@ -170,7 +171,8 @@
 		for (var/datum/category_item/player_setup_item/PI in items)
 			PI.save_preferences(S)
 	else
-		handle_sql_saving(SQL_PREFERENCES)
+		if (sql_role == SQL_PREFERENCES)
```

Comment:
This is unnecessary, because handle_sql_saving does the exact same thing. Same for loading. :ree: